src/language-server/settings.ts

```diff
@@ -50,13 +58,25 @@ export default class Settings extends AutoDisposable {
   readonly projectConfigChangeStream: EventStream<Change<ProjectConfigInfo>>;
   private readonly fireProjectConfigChange:
       (change: Change<ProjectConfigInfo>) => void;
+  /**
+   * A promise that resolves once the initial configuration has probably been
+   * received from the client.
+   *
+   * There's no way to know for certain though, because it's asynchronous, so
+   * after a timeout this will resolve either way.
+   */
+  readonly ready: Promise<void>;
   constructor(
       connection: IConnection, fileSynchronizer: FileSynchronizer,
       converter: AnalyzerLSPConverter) {
     super();
     const {fire, stream} = EventStream.create<Change<SettingsJson>>();
     this.changeStream = stream;
+    let resolveReady: () => void;
```

Comment:
the goal of these lines is to resolve the `ready` promise either at the first `connection.onDidChangeConfiguration` or at most in 1 second, correct?

Reply:
Correct!